machine-learning/sagemaker-edge-manager/artifacts/com.greengrass.SageMakerEdgeManager.ImageClassification/1.0.0/image_classification/prediction_utils.py

```diff
@@ -0,0 +1,151 @@
+from ast import literal_eval
+from datetime import datetime
+from json import dumps
+
+import config_utils
+import cv2
+import IPCUtils as ipc_utils
+import numpy as np
+from agent_pb2 import (
+    ListModelsRequest,
+    LoadModelRequest,
+    PredictRequest,
+    Tensor,
+    TensorMetadata,
+    UnLoadModelRequest,
+)
+
+config_utils.logger.info("Using np from '{}'.".format(np.__file__))
+config_utils.logger.info("Using cv2 from '{}'.".format(cv2.__file__))
+
+# Read labels file
+with open(config_utils.LABELS_FILE, "r") as f:
+    labels = literal_eval(f.read())
+
+
+def transform_image(im):
+    if len(im.shape) == 2:
+        height, width = im.shape[:2]
+        im = np.expand_dims(im, axis=2)
+        nchannels = 1
+    elif len(im.shape) == 3:
+        height, width, nchannels = im.shape[:3]
+    else:
+        raise Exception("Unknown image structure")
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Using `Exception` and `BaseException` can make your code prone to errors and difficult to maintain. Instead, we recommend using one of the [Built-in Exceptions](https://docs.python.org/3/library/exceptions.html) or creating a custom exception class that is derived from `Exception` or one of its subclasses.